scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -1355,12 +1362,19 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
         next()
         t = atPos(t, auto)(Term.Match(t, inBracesOrNil(caseClauses())))
       }
-      // in order to allow anonymous functions as statements (as opposed to expressions) inside
-      // templates, we have to disambiguate them from self type declarations - bug #1565
-      // The case still missed is unparenthesized single argument, like "x: Int => x + 1", which
-      // may be impossible to distinguish from a self-type and so remains an error.  (See #1564)
+
+      t = t match {
+        case q1 @ Term.Quasi(r1, q2 @ Term.Quasi(r2, tree)) =>
+          atPos(q1, q1)(Term.Tuple(List(atPos(q2, q2)(Term.Quasi(q1.rank, q1.tree)))))
+        case _ => t
+      }
+
+      lazy val isInBraces = t.tokens.nonEmpty && t.tokens.head.is[`(`] && t.tokens.last.is[`)`]
       def lhsIsTypedParamList() = t match {
```

Comment:
At the moment name is correct. As far as I understood, fix for tuples with not only Term.Ascribe (like `(x: X, y, z: Z)`) is not intended to be included in this PR.